### PR TITLE
[DOC] comments for readability of `pyproject.toml`

### PR DESCRIPTION
**Pull request by fkiraly**
I added comments for readability of `pyproject.toml`, in the file:

* semantic ordering and commenting of dependency sets
* comments in headers re contributor credits and governance
* explanations for core dep set

### CODE REVIEW

**Comment by fkiraly** [comment]
FYI @yarnabrina, this PR can also e used as a test case for the `pyproject` CI trigger.

File: pyproject.toml [hunk 1]
```diff
@@ -46,19 +50,32 @@ classifiers = [
...
+# this set should be kept minimal!
 dependencies = [
-  "numpy<1.27,>=1.21",
-  "packaging",
-  "pandas<2.2.0,>=1.1",
-  "scikit-base<0.7.0",
-  "scikit-learn<1.4.0,>=0.24",
-  "scipy<2.0.0,>=1.2",
+  "numpy<1.27,>=1.21",  # required for framework layer and base class logic
+  "packaging",  # for estimator specific dependency parsing
```
Comment (yarnabrina) [inline pyproject.toml:57]: Does not correspond to PR title, but may be worth to add lower and upper bounds based on previous discussions?

File: pyproject.toml [hunk 2]
```diff
@@ -46,19 +50,32 @@ classifiers = [
...
+  "pandas<2.2.0,>=1.1",  # pandas is the main in-memory data container
+  "scikit-base<0.7.0",  # base module for sklearn compatible base API
```
Comment (yarnabrina) [inline pyproject.toml:59]: Same as above, suggesting lower bound.

**Review by yarnabrina — COMMENTED** [review]
Looks fine, and very useful for future reference. However, probably (not sure) comments are not supported by pyproject-fmt and it also forces alphabetical sorting of extras, so may be it's better to skip it altogether given so many restrictions.

Other than two minor comments below, can I suggest to consider keeping docs and tests separately from other CI extras? They are much more standard and common and new contributors may use those.

File: pyproject.toml (hunk 1 above)
Comment (fkiraly) [reply pyproject.toml:57]: what do you mean? All I added were the comments. If the deps change, that's a mistake on my side, but I can see no changes?

File: pyproject.toml (hunk 2 above)
Comment (fkiraly) [reply pyproject.toml:59]: this PR scope is only adding comments and possibly changing the order, but not the comment of `pyproject.toml`.

File: pyproject.toml (hunk 1 above)
Comment (yarnabrina) [reply pyproject.toml:57]: That's what I meant by "Does not correspond to PR title". Resolving both comments.

**Comment by fkiraly** [comment]
ok, I separated out `docs` and `tests`

**Review by yarnabrina — APPROVED** [review]

**Pull request merged**